devito/tools/utils.py

```diff
@@ -161,8 +162,7 @@ def filter_ordered(elements, key=None):
     seen = set()
     if key is None:
         try:
-            unordered, inds = np.unique(elements, return_index=True)
-            return unordered[np.argsort(inds)].tolist()
+            return pd.unique(elements.ravel()).tolist()
```

Comment:
Putting back my comment since seems to have not saved it. THis shouldn't use `ravel` since we can have 
- lists
- array of the form `[ (1,2), (3,4), (1,2)]`
as input and `ravel` doesn't work on lists and would flatten the tuples in the second case.